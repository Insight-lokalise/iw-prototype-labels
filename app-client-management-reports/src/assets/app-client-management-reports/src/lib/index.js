export { formatDate, parseDate } from './date'
