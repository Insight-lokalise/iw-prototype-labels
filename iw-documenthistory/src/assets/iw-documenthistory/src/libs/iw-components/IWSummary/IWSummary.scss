@charset "utf-8";
@import 'iw-styles/src/iw-settings';


$iw-summary-padding: rem-calc(10);
$iw-summary-margin: rem-calc(15);


.iw-summary {
  background-color: white;

  &-ewr-modal__body {
    padding: $global-spacing-unit-small !important;
  }
}

  .iw-summary__header {
    padding: $iw-summary-padding 0;
    border-bottom: 1px solid $brand-lightgray;
    color: $brand-darkgray;
  }

    .iw-summary__title {
      margin: 0;
    }

  .iw-summary__totals {
    margin-top: $iw-summary-margin;
  }

    .iw-summary__row {
      padding-top: $iw-summary-padding;
      padding-bottom: $iw-summary-padding;
    }

      .iw-summary__label {
        font-weight: $global-weight-bold;

        .iw-summary__row--total & {
          font-size: rem-calc(18) !important;
        }

      }

      .iw-currency__amount {

        .iw-summary__row--total & {
          font-size: rem-calc(18) !important;
        }

      }

    .iw-summary__hline {
      border-color: $brand-lightgray;
      margin: $iw-summary-padding 0;
    }

    .iw-summary__footnote {
      margin: $iw-summary-padding 0 $iw-summary-padding / 2;
      font-size: .875em;
      line-height: 1.4;
    }
