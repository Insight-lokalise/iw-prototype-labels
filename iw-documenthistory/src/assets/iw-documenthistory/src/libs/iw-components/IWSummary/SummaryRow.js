import React from 'react'
import PropTypes from 'prop-types'
import cn from 'classnames'

export default function SummaryRow({ className, title, priceElement }) {
  return (
    <div className={cn('row is-collapse-child', className)}>
      <div className="columns iw-summary__label">{title}</div>
      <div className="columns shrink iw-summary__value">{priceElement}</div>
    </div>
  )
}

SummaryRow.propTypes = {
  className: PropTypes.string,
  priceElement: PropTypes.node.isRequired,
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
}

SummaryRow.defaultProps = {
  className: '',
}
