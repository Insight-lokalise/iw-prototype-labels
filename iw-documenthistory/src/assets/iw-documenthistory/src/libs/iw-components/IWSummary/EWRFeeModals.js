import React from 'react'

import { t } from '@insight/toolkit-utils/lib/labels'
import { IWAnchor } from '../IWAnchor/IWAnchor'

export function CanadaEWRFeeModalBody(props) {
  const paragraphOne = (
    <p>
      {t(
        'The Electronic Products Recycling Association (EPRA) is an industry-led, not-for-profit organization that operates regulated recycling programs across Canada. They ensure that end-of-life electronics are handled in a safe, secure and environmentally-sound manner. Please refer to the following links for more information about the recycling program and EWR fees in the province you are shipping to.'
      )}
    </p>
  )
  const paragraphTwo = (
    <p>
      <IWAnchor target="blank" href="https://www.albertarecycling.ca/">
        {t('Alberta (operated by Alberta Recycling Management Authority)')}
      </IWAnchor>
      <br />
      <IWAnchor target="blank" href="https://www.recyclemyelectronics.ca/bc/">
        {t('British Columbia')}
      </IWAnchor>
      <br />
      <IWAnchor target="blank" href="https://www.recyclemyelectronics.ca/mb/">
        {t('Manitoba')}
      </IWAnchor>
      <br />
      <IWAnchor target="blank" href="https://www.recyclemyelectronics.ca/nb/">
        {t('New Brunswick')}
      </IWAnchor>
      <br />
      <IWAnchor target="blank" href="https://www.recyclemyelectronics.ca/nl">
        {t('Newfoundland and Labrador')}
      </IWAnchor>
      <br />
      <IWAnchor target="blank" href="https://www.recyclemyelectronics.ca/ns/">
        {t('Nova Scotia')}
      </IWAnchor>
      <br />
      <IWAnchor target="blank" href="https://www.recycleyourelectronics.ca/">
        {t('Ontario')}
      </IWAnchor>
      <br />
      <IWAnchor target="blank" href="https://www.recyclemyelectronics.ca/pei/">
        {t('Prince Edward Island')}
      </IWAnchor>
      <br />
      <IWAnchor target="blank" href={props.locale === 'fr_CA' ? 'https://www.recyclermeselectroniques.ca/qc/' : 'https://www.recyclemyelectronics.ca/qc/'}>
        {t('Quebec')}
      </IWAnchor>
      <br />
      <IWAnchor target="blank" href="https://www.recyclemyelectronics.ca/sk/">
        {t('Saskatchewan')}
      </IWAnchor>
      <br />
    </p>
  )

  return (
    <div className="column">
      {paragraphOne}
      {paragraphTwo}
    </div>
  )
}

export function CaliforniaEWRFeeModalBody() {
  const paragraphOne = 'Electronic Waste Recycling Act of 2003'
  const paragraphTwo = 'On September 24, 2003, California passed legislation establishing a funding system for the collection and recycling of certain electronic wastes. According to the Electronic Waste Recycling Act of 2003, retailers and direct sellers must impose a fee at the time of purchase on certain electronic items. Insight will begin assessing this fee on all orders shipping to California starting on January 1, 2005. For the full text of the law, visit'
  const paragraphThree = 'Insight is committed to adhering to this legislation by assessing the appropriate fee to all orders containing affected products shipping into California starting on January 1, 2005. Affected products include but are not limited to: computer monitors both CRT and LCD, notebook computers, projectors, televisions, any video display devices with a screen size 4 inches or larger (phones, fax machines, cameras). The fees range from $6 and up per unit.'
  const paragraphFive = 'We value your partnership and look forward to continually growing our business with you.'
  return (
      <div className="iw-summary-ewr-modal__body" >
          <p>{t(paragraphOne)}</p>
          <p>
            {t(paragraphTwo)}
            <IWAnchor target="blank" href="https://www.calrecycle.ca.gov/Electronics/">https://www.calrecycle.ca.gov/Electronics/</IWAnchor>.
          </p>
          <p>{t(paragraphThree)}</p>
          <p>{t('Any questions regarding this fee should be directed to')}<IWAnchor target="blank" href="mailto:CARecycling@Insight.com?subject=EWR%Question">CARecycling@Insight.com</IWAnchor>.</p>
          <p>{t(paragraphFive)}</p>
      </div>
  )
}