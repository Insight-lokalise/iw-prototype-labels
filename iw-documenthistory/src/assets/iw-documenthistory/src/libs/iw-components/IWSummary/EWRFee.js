import React from 'react'
import PropTypes from 'prop-types'

import Currency from '@insight/toolkit-react/lib/Currency/Currency'
import { t, getCurrentLocale } from '@insight/toolkit-utils'

import { IWTooltipModal } from '../IWModal'
import SummaryRow from './SummaryRow'

export default function EWRFee({ isCanada, ewrFee, currency }) {
  const ewrText = t('EWR')
  const ewrTitle = t('Electronic waste recycling')
  const locale = getCurrentLocale("insight_current_locale");

  return (
    <SummaryRow
      className="iw-summary__row"
      title={
        <span>
          <abbr title={ewrTitle}>{`${ewrText} `}</abbr>
          <IWTooltipModal
            isCanada={isCanada}
            locale={locale}
          />
        </span>
      }
      priceElement={<Currency value={ewrFee} currencyCode={currency} />}
    />
  )
}

EWRFee.propTypes = {
  currency: PropTypes.string.isRequired,
  ewrFee: PropTypes.number.isRequired,
  isCanada: PropTypes.bool.isRequired,
}
