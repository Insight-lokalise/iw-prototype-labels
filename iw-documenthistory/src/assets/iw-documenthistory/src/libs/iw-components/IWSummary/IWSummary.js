import React from "react";
import PropTypes from "prop-types";
import cn from "classnames";

import Currency from "@insight/toolkit-react/lib/Currency/Currency";
import { t } from "@insight/toolkit-utils/lib/labels";
import SummaryRow from "./SummaryRow";
import EWRFee from "./EWRFee";
import IWHelpIcon from "./../IWTooltip/IWHelpIcon";

export const IWSummary = ({
  carrierDescription,
  children,
  className,
  currency,
  ewrFee,
  gstHstTaxCost,
  isCanada,
  pstTaxCost,
  shippingCost,
  isXD,
  showBlankTax,
  showCarrierDescription,
  showEWR,
  showShippingHelpText,
  showTax,
  subTotal,
  taxCost,
  usedThirdPartyCarrier,
  isEMEA,
}) => {
  const shippingText = usedThirdPartyCarrier
    ? t("Charge Account")
    : t("Estimated shipping");
  const showPstTax = showTax && pstTaxCost > 0;
  const showGstHstTax = showTax && gstHstTaxCost > 0;

  // only show one type of tax, and default to regular tax.
  const shouldShowTax = showTax && !isCanada;

  // we calculate the total based on the showX props so that consumerse do not
  // need to wrap duplicate their showX logic on the cost props.
  const sumTotal =
    subTotal +
    (isXD ? 0 : shippingCost) +
    (shouldShowTax ? (showBlankTax ? 0 : taxCost) : 0) +
    (showEWR ? ewrFee : 0) +
    (showPstTax ? pstTaxCost : 0) +
    (showGstHstTax ? gstHstTaxCost : 0);

  const taxLabel = isEMEA ? (
    <>
      {t("VAT")}&nbsp;
      <span className="hide-for-print">
        <IWHelpIcon
          tooltip={t(
            "Goods and services are invoiced at the price prevailing at time of acceptance of order. VAT is charged at the rate applicable at the time of invoicing or otherwise in accordance with the law."
          )}
        />
      </span>
    </>
  ) : (
    t("Estimated tax")
  );

  return (
    <div className={cn(className)}>
      <div className="iw-summary">
        <div className="row iw-summary__header align-middle is-collapse-child">
          <h3 className="columns iw-summary__title">{t("Summary")}</h3>
        </div>
        <div className="iw-summary__totals">
          <SummaryRow
            className="iw-summary__row iw-summary__row--subtotal"
            title={t("Subtotal")}
            priceElement={<Currency value={subTotal} currencyCode={currency} />}
          />
          <SummaryRow
            className="iw-summary__row iw-summary__row--shipping"
            title={
              <span>
                {showShippingHelpText && (
                  <span className="hide-for-print">*</span>
                )}
                {shippingText}
                {showCarrierDescription && carrierDescription && (
                  <span>
                    <p className="no-margin-bot">({carrierDescription})</p>
                  </span>
                )}
              </span>
            }
            priceElement={
              isXD ? (
                <span>--</span>
              ) : (
                <Currency value={shippingCost} currencyCode={currency} />
              )
            }
          />
          {showEWR && ewrFee > 0 && (
            <EWRFee currency={currency} ewrFee={ewrFee} isCanada={isCanada} />
          )}
          {shouldShowTax && (
            <SummaryRow
              className="iw-summary__row iw-summary__row--tax"
              title={taxLabel}
              priceElement={
                showBlankTax ? (
                  <span data-testid="empty-tax">--</span>
                ) : (
                  <Currency value={taxCost} currencyCode={currency} />
                )
              }
            />
          )}
          {showPstTax && isCanada &&(
            <SummaryRow
              className="iw-summary__row iw-summary__row--tax"
              title={t("PST / QST est.")}
              priceElement={
                showBlankTax ? (
                  <span data-testid="empty-psttax">--</span>
                ) : (
                  <Currency value={pstTaxCost} currencyCode={currency} />
                )
              }
            />
          )}
          {showTax && isCanada && (
            <SummaryRow
              className="iw-summary__row iw-summary__row--tax"
              title={t("GST / HST est.")}
              priceElement={
                showBlankTax ? (
                  <span data-testid="empty-gsttax">--</span>
                ) : (
                  <Currency value={gstHstTaxCost} currencyCode={currency} />
                )
              }
            />
          )}
          <div className="row is-collapse-child">
            <div className="columns u-padding-none">
              <hr className="iw-summary__hline" />
            </div>
          </div>
          <SummaryRow
            className={"iw-summary__row iw-summary__row--total"}
            title={t("Total")}
            priceElement={<Currency value={sumTotal} currencyCode={currency} />}
          />
        </div>
        {!!children && children}
        {showShippingHelpText && (
          <div className="row is-collapse-child hide-for-print">
            <div className="columns">
              <p className="iw-summary__footnote">
                {t(
                  "*Shipping rates are calculated based on your default information. Other shipping rate options can be viewed and selected on the next page."
                )}
              </p>
            </div>
          </div>
        )}
      </div>
    </div>
  );
};

IWSummary.propTypes = {
  carrierDescription: PropTypes.string,
  className: PropTypes.string,
  currency: PropTypes.string.isRequired,
  ewrFee: PropTypes.number,
  gstHstTaxCost: PropTypes.number,
  isCanada: PropTypes.bool,
  pstTaxCost: PropTypes.number,
  shippingCost: PropTypes.number.isRequired,
  isXD: PropTypes.bool,
  showBlankTax: PropTypes.bool,
  showCarrierDescription: PropTypes.bool,
  showEWR: PropTypes.bool,
  showShippingHelpText: PropTypes.bool,
  showTax: PropTypes.bool,
  subTotal: PropTypes.number.isRequired,
  taxCost: PropTypes.number,
  usedThirdPartyCarrier: PropTypes.bool,
  isEMEA: PropTypes.bool,
};

IWSummary.defaultProps = {
  carrierDescription: null,
  className: null,
  ewrFee: null,
  gstHstTaxCost: null,
  isCanada: null,
  pstTaxCost: null,
  isXD: null,
  showBlankTax: null,
  showCarrierDescription: null,
  showEWR: null,
  showShippingHelpText: null,
  showTax: null,
  taxCost: null,
  usedThirdPartyCarrier: null,
  isEMEA: null,
};

export default IWSummary;
