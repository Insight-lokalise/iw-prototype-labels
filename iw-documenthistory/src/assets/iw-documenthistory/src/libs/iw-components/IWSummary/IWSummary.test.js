import React from 'react'
import { fireEvent } from '@testing-library/react'
import { render } from 'test-utils'

import IWSummary from './IWSummary'
import { t, l } from '@insight/toolkit-utils/lib/labels'

jest.mock('@insight/toolkit-utils/lib/labels', ()=>{
  return {
    l: jest.fn(() => 'en_US'),
    t: jest.fn(string => string)
  }
})


let wrapper
const costProps = {
  subTotal: 10000,
  shippingCost: 1000,
  taxCost: 100,
  ewrFee: 10,
  pstTaxCost: 1,
  gstHstTaxCost: 0.1,
  isCanada: false,
}

describe('IWSummary', () => {
  describe('renders correctly', () => {
    afterEach(() => wrapper.unmount())

    it('renders something', () => {
      wrapper = render(<IWSummary {...costProps} showTax showEWR showPstTax showGstHstTax currency={'USD'} />)
      const { asFragment} = wrapper
      expect(asFragment()).toMatchSnapshot()
    })

    it('renders children', () => {
      wrapper = render(
        <IWSummary {...costProps} currency={'USD'}>
          <span data-testid="render-check" className="render-check" />
        </IWSummary>
      )
      const {queryByTestId} = wrapper
      expect(queryByTestId('render-check')).toBeInTheDocument()
    })

    it('renders blank tax', () => {
      wrapper = render(<IWSummary {...costProps} showTax showBlankTax currency={'USD'} />)
      const {queryByTestId} = wrapper
      expect(queryByTestId('empty-psttax')).toBeInTheDocument()
      expect(queryByTestId('empty-gsttax')).toBeInTheDocument()
    })

    it('renders carrierDescription', () => {
      wrapper = render(
        <IWSummary {...costProps} showCarrierDescription carrierDescription={'description'} currency={'USD'} />
      )
      const {getByText} = wrapper
      expect(getByText('(description)')).toBeInTheDocument()
    })

    it('handles 3rd party carriers:', () => {
      wrapper = render(<IWSummary {...costProps} usedThirdPartyCarrier currency={'USD'} />)
      const {getByText} = wrapper
      expect(getByText('Charge Account')).toBeInTheDocument()
    })
  })

  describe('calculates the correct displayed total', () => {
    afterEach(() => wrapper.unmount())

    it('excludes hidden values', () => {
      // only subTotal and shipping costs are included by default
      wrapper = render(<IWSummary {...costProps} currency={'USD'} />)
      const {getByText} = wrapper
      expect(getByText('$11,000.00')).toBeInTheDocument()
    })
  })
})
