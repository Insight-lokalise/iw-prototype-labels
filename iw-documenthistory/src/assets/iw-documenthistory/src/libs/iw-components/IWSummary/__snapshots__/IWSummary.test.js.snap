// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IWSummary renders correctly renders something 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <div
      class="iw-summary"
    >
      <div
        class="row iw-summary__header align-middle is-collapse-child"
      >
        <h3
          class="columns iw-summary__title"
        >
          Summary
        </h3>
      </div>
      <div
        class="iw-summary__totals"
      >
        <div
          class="row is-collapse-child iw-summary__row iw-summary__row--subtotal"
        >
          <div
            class="columns iw-summary__label"
          >
            Subtotal
          </div>
          <div
            class="columns shrink iw-summary__value"
          >
            <span
              aria-label="USD$10,000.00 "
              class="c-currency"
              tabindex="0"
            >
              <span
                class="c-currency__code"
              >
                USD 
              </span>
              <span
                class="c-currency__value"
              >
                $10,000.00
              </span>
            </span>
          </div>
        </div>
        <div
          class="row is-collapse-child iw-summary__row iw-summary__row--shipping"
        >
          <div
            class="columns iw-summary__label"
          >
            <span>
              Shipping estimate
            </span>
          </div>
          <div
            class="columns shrink iw-summary__value"
          >
            <span
              aria-label="USD$1,000.00 "
              class="c-currency"
              tabindex="0"
            >
              <span
                class="c-currency__code"
              >
                USD 
              </span>
              <span
                class="c-currency__value"
              >
                $1,000.00
              </span>
            </span>
          </div>
        </div>
        <div
          class="row is-collapse-child iw-summary__row"
        >
          <div
            class="columns iw-summary__label"
          >
            <span>
              <abbr
                title="Electronic waste recycling"
              >
                EWR 
              </abbr>
              <span
                class="ion-help-circled help-icon__icon hide-for-print"
              />
            </span>
          </div>
          <div
            class="columns shrink iw-summary__value"
          >
            <span
              aria-label="USD$10.00 "
              class="c-currency"
              tabindex="0"
            >
              <span
                class="c-currency__code"
              >
                USD 
              </span>
              <span
                class="c-currency__value"
              >
                $10.00
              </span>
            </span>
          </div>
        </div>
        <div
          class="row is-collapse-child iw-summary__row iw-summary__row--tax"
        >
          <div
            class="columns iw-summary__label"
          >
            PST / QST estimate
          </div>
          <div
            class="columns shrink iw-summary__value"
          >
            <span
              aria-label="USD$1.00 "
              class="c-currency"
              tabindex="0"
            >
              <span
                class="c-currency__code"
              >
                USD 
              </span>
              <span
                class="c-currency__value"
              >
                $1.00
              </span>
            </span>
          </div>
        </div>
        <div
          class="row is-collapse-child iw-summary__row iw-summary__row--tax"
        >
          <div
            class="columns iw-summary__label"
          >
            GST / HST estimate
          </div>
          <div
            class="columns shrink iw-summary__value"
          >
            <span
              aria-label="USD$0.10 "
              class="c-currency"
              tabindex="0"
            >
              <span
                class="c-currency__code"
              >
                USD 
              </span>
              <span
                class="c-currency__value"
              >
                $0.10
              </span>
            </span>
          </div>
        </div>
        <div
          class="row is-collapse-child"
        >
          <div
            class="columns u-padding-none"
          >
            <hr
              class="iw-summary__hline"
            />
          </div>
        </div>
        <div
          class="row is-collapse-child iw-summary__row iw-summary__row--total"
        >
          <div
            class="columns iw-summary__label"
          >
            Total
          </div>
          <div
            class="columns shrink iw-summary__value"
          >
            <span
              aria-label="USD$11,011.10 "
              class="c-currency"
              tabindex="0"
            >
              <span
                class="c-currency__code"
              >
                USD 
              </span>
              <span
                class="c-currency__value"
              >
                $11,011.10
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
